### PR TITLE
Clear history listener and timeouts on unmount

### PR DESCRIPTION
**Pull request by tamorim**

### CODE REVIEW

**Comment by brenoc** [comment]
https://github.com/vtex-apps/theme/blob/fix/default-template/src/components/DefaultTemplate/DefaultTemplate.js#L58-L62

Why we are doing this? Can we assume the ContextStore is not loading anymore just because the store changed? I think this is far fetched.

Maybe this is causing the bug of clicking like an absolute madman.

**Comment by tamorim** [comment]
@brenoc Hmn, I think that only the `return` is unecessary there. What do you think?
This way any change get through the 200ms timeout, canceling the one before.
Or we could treat the `false` on `loading` to not go through a timeout.

**Comment by tamorim** [comment]
@brenoc But I'm not convinced that the loading is responsible for the bug with multiple clicks, I will change and see

**Comment by tamorim** [comment]
@brenoc Yup, it made no difference for that bug... also found another one!
If you click to go to a path you already are, the page goes blank :/
I'll treat this stuff on the SDK.

**Comment by tamorim** [comment]
@brenoc Done!

File: src/components/DefaultTemplate/DefaultTemplate.js [hunk 1]
```diff
@@ -46,19 +46,27 @@ class DefaultTemplate extends React.Component {
...
     if (this.loadTimeout) {
       clearTimeout(this.loadTimeout);
       this.loadTimeout = null;
-      return;
     }
 
-    this.loadTimeout = setTimeout(() => {
-      this.setState({ loading: ContextStore.get('loading') });
-    }, 200);
+    if (!loading) {
```
Comment (brenoc) [inline src/components/DefaultTemplate/DefaultTemplate.js]: Maybe you could do:

```js
if (this.state.loading && !loading)  {
  this.setState({ loading });
  return;
}
```

**Comment by tamorim** [comment]
@brenoc Nice catch! I forgot the `return`there hehe
Also added the state comparison the prevent useless rerendering :+1:

**Pull request merged**